### PR TITLE
Refactor carts and orders

### PR DESCRIPTION
**Pull request by lauraluiz**
More stuff extracted from checkout PR.
This one refactors all cart and order components, so that for every view with some parts of the carts or orders, the components can be shared. So in short: following [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

### CODE REVIEW

File: src/mixins/cartMixin.js [hunk 1]
```diff
@@ -130,6 +74,30 @@ export default {
...
+          version: this.me.activeCart?.version,
+          locale: this.$store.state.locale,
+        },
+        update: (store) => {
+          const data = store.readQuery({ query: BASIC_CART_QUERY });
+          data.me.activeCart = null;
+          store.writeQuery({ query: BASIC_CART_QUERY, data });
+        },
+      });
+    },
```
Comment (lauraluiz) [inline src/mixins/cartMixin.js]: Well and this feature sneaked in, from the checkout ;)
But if I remove it, I'll easily run into conflicts later, so I hope you forgive me 🙏

File: src/components/Cart.gql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      name(locale: $locale)
+    }
+  }
+  shippingAddress {
+    ...AddressFields
+  }
+  billingAddress {
+    ...AddressFields
+  }
+}
```
Comment (lauraluiz) [inline src/components/Cart.gql]: Extracting this "common fields" queries into their own GQL fields will make them (hopefully) easier to maintain in the future.

File: src/components/Address.gql [hunk 3]
```diff
@@ -1,4 +1,4 @@
-fragment BaseAddress on Address {
+fragment AddressFields on Address {
```
Comment (lauraluiz) [inline src/components/Address.gql]: Trying to give it a more generic name. "Base" sounded as there could be an extended version.

File: src/mixins/cartMixin.js (hunk 1 above)
Comment (zeina-idris) [reply src/mixins/cartMixin.js]: haha I forgive you!

**Pull request merged**